packages/snaps-rpc-methods/src/permitted/requestPermissions.ts

```diff
@@ -0,0 +1,123 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type {
+  PermissionConstraint,
+  PermittedHandlerExport,
+  RequestedPermissions,
+  SubjectPermissions,
+} from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  RequestPermissionsParams,
+  RequestPermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { Snap } from '@metamask/snaps-utils';
+import { hasProperty, isPlainObject } from '@metamask/utils';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import deepEqual from 'fast-deep-equal';
+
+import { processSnapPermissions } from '../permissions';
+import type { MethodHooksObject } from '../utils';
+
+export type RequestPermissionsHooks = {
+  getOriginSnap: () => Snap | undefined;
+  /**
+   * @returns Requested permission result.
+   */
+  requestPermissions: (
+    dynamicPermissions: RequestedPermissions,
+  ) => Promise<SubjectPermissions<PermissionConstraint>>;
```

Comment:
This type does not match the return value of `requestPermissions` as `requestPermissions` returns an array: https://github.com/MetaMask/snaps/blob/main/packages/snaps-rpc-methods/src/permitted/requestSnaps.ts#L60-L67

Reply:
Updated.